openbas-api/src/main/java/io/openbas/rest/team/ExerciseTeamApi.java

```diff
@@ -0,0 +1,45 @@
+package io.openbas.rest.team;
+
+import io.openbas.database.model.Team;
+import io.openbas.rest.helper.RestBehavior;
+import io.openbas.rest.team.output.TeamOutput;
+import io.openbas.service.TeamService;
+import io.openbas.telemetry.Tracing;
+import io.openbas.utils.pagination.SearchPaginationInput;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotBlank;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import static io.openbas.database.model.User.ROLE_USER;
+import static io.openbas.database.specification.TeamSpecification.contextual;
+import static io.openbas.database.specification.TeamSpecification.fromExercise;
+import static io.openbas.rest.exercise.ExerciseApi.EXERCISE_URI;
+
+@RequiredArgsConstructor
+@RestController
+@Secured(ROLE_USER)
+public class ExerciseTeamApi extends RestBehavior {
+
+  private final TeamService teamService;
+
+  @PostMapping(EXERCISE_URI + "/{exerciseId}/teams/search")
+  @PreAuthorize("isExerciseObserver(#exerciseId)")
+  @Transactional(readOnly = true)
+  @Tracing(name = "Paginate teams for exercise", layer = "api", operation = "POST")
+  public Page<TeamOutput> searchTeams(
+      @PathVariable @NotBlank final String exerciseId,
+      @RequestBody @Valid SearchPaginationInput searchPaginationInput) {
+    final Specification<Team> teamSpecification = contextual(false).or(fromExercise(exerciseId).and(contextual(true)));
+    return this.teamService.teamPagination(searchPaginationInput, teamSpecification);
```

Comment:
should be final Specification<Team> teamSpecification =fromExercise(exerciseId); 
nevermind about contextual teams because the path of your request is already /{exerciseId}/teams/search 

What do you think ?

Reply:
We are looking teams that can be set to this exercise
We have two cases:
1. platform team -> not contextual
2. contextual team in this exercise -> contextual & exerciseId